src/main/java/programminglife/gui/controller/GuiLoadBookmarkController.java

```diff
@@ -50,53 +47,61 @@ public void initialize() {
      * Checks whether the user has selected a bookmark.
      * @return True if selected, false otherwise.
      */
-    private boolean checkBookmarkSelection() {
-        if (tblBookmark.getSelectionModel().getSelectedItem() == null) {
-            Alert alert = new Alert(Alert.AlertType.INFORMATION);
-            alert.setTitle("No bookmark selected");
-            alert.setContentText("Please select a bookmark to open");
-            alert.setHeaderText(null);
-            alert.show();
-            return false;
+    private Bookmark checkBookmarkSelection() {
+        Bookmark bookmark;
+        for (TableView<Bookmark> tableView : tableViews) {
+            if (tableView.getSelectionModel().getSelectedItem() != null) {
+                bookmark = tableView.getSelectionModel().getSelectedItem();
+                return bookmark;
+            }
         }
-        return true;
+        Alerts.warning("No bookmark selected").show();
+        return null;
     }
 
     /**
      * Initializes the buttons in the window.
      */
     private void initButtons() {
         btnOpenBookmark.setOnAction(event -> {
-            if (checkBookmarkSelection()) {
-                Bookmark bookmark = tblBookmark.getSelectionModel().getSelectedItem();
+            Bookmark bookmark = checkBookmarkSelection();
+            if (bookmark != null) {
+                if (guiController.getFile() == null
+                        || !bookmark.getPath().equals(guiController.getFile().getAbsolutePath())) {
+                    File file = new File(bookmark.getPath());
+                    try {
+                        guiController.setFile(file);
+                        guiController.openFile(file);
+                    } catch (IOException | UnknownTypeException e) {
```

Comment:
I like this but when i tested it It did not give an alert but gave me a FileNotFoundException.